### PR TITLE
feat: plugin instance envs

### PR DESCRIPTION
**Pull request by CodeBeater**

### CODE REVIEW

File: sdk/index.ts [hunk 1]
```diff
@@ -17,6 +20,8 @@ declare namespace host {
   export function vector_embedding_search(request: u32): u32;
   export function kv_get_user_string(request: u32): u32;
   export function kv_set_user_string(request: u32): void;
+  export function plugin_env_get_string(request: u32): u32;
+  export function plugin_env_set_string(request: u32): void;
```
Comment (rellfy) [inline sdk/index.ts:24]: Did you forget to remove this one?

File: sdk/index.ts [hunk 2]
```diff
@@ -291,6 +296,29 @@ export class UserKvStorage {
...
+      request,
+      HostPluginEnvGetStringRequest.encode,
+    );
+    const responsePtr = host.plugin_env_get_string(writeBufferToPr(requestBytes));
+    const response = Protobuf.decode<HostPluginEnvGetStringResponse>(
+      readBufferFromPtr(responsePtr),
+      HostPluginEnvGetStringResponse.decode,
+    );
+    
+    return response.value ? response.value : "";
```
Comment (rellfy) [inline sdk/index.ts:318]: (Keeping in mind https://github.com/asterai-io/as-proto/issues/1)
```suggestion
    return response.value || "";
```

File: sdk/index.ts (hunk 1 above)
Comment (CodeBeater) [reply sdk/index.ts]: Yes, addressed.

**Review by rellfy — COMMENTED** [review]
Also bump the version of the SDK pls

File: sdk/index.ts [hunk 3]
```diff
@@ -291,6 +294,29 @@ export class UserKvStorage {
   }
 }
 
+export class PluginEnvStorage {
+  private readonly plugin_id: string;
+
+  public constructor(plugin_id: string) {
+    this.plugin_id = plugin_id;
+  }
```
Comment (rellfy) [inline sdk/index.ts:302]: This shouldn't be required

File: sdk/index.ts [hunk 4]
```diff
@@ -291,6 +294,29 @@ export class UserKvStorage {
...
+      request,
+      HostPluginEnvGetStringRequest.encode,
+    );
+    const responsePtr = host.plugin_env_get_string(writeBufferToPr(requestBytes));
+    const response = Protobuf.decode<HostPluginEnvGetStringResponse>(
+      readBufferFromPtr(responsePtr),
+      HostPluginEnvGetStringResponse.decode,
+    );
+    
+    return response.value || "";
```
Comment (rellfy) [inline sdk/index.ts:316]: This should probably return `string  | null` instead.

Comment (CodeBeater) [reply sdk/index.ts]: Addressed

File: sdk/index.ts (hunk 3 above)
Comment (CodeBeater) [reply sdk/index.ts]: Addressed

File: examples/wolfram/plugin.asterai.proto [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import "node_modules/@asterai/sdk/protobuf/asterai.proto";
+
+service Wolfram {
+  //provides simple textual answers for any valid WolframAlpha query in natural language. is able to also provide the market price for commodity metals such as gold,
+  //silver and palladium. this method can only provide the answer for one question or one price at a time.
+  rpc textQuery(TextQueryRequest) returns (TextQueryResponse) {}
+}
+
+message TextQueryRequest {
+  //the url-encoded query to send to WolframAlpha
```
Comment (rellfy) [inline examples/wolfram/plugin.asterai.proto:11]: Please add a spacing here and begin the comment with a capital letter, just for consistency.

**Review by rellfy — APPROVED** [review]

**Review by rellfy — APPROVED** [review]

File: examples/wolfram/plugin.asterai.proto (hunk 5 above)
Comment (CodeBeater) [reply examples/wolfram/plugin.asterai.proto]: Addressed

**Pull request merged**